components/ide-proxy/Dockerfile

```diff
@@ -13,6 +13,9 @@ RUN for FILE in `ls /bin/gitpod-local-companion*`;do \
   gzip -v -f -9 -k "$FILE"; \
 done
 
+RUN mkdir -p static/code
+RUN curl -o static/code/marketplace.json "https://raw.githubusercontent.com/open-vsx/publish-extensions/master/extension-control/extensions.json"
```

Comment:
@mustard-mh do you see anything obvious about why my approach with downloading the file does not work? It seems the file and its containing folder is missing in the resulting docker image 🤔

Reply:
Not sure if you aware, but if we write this curl inside Dockerfile, then it will only changed when this components is rebuild. 

We could add a line before download like `ENV TRIGGER_REBUILD 1` when you want to force update it, we could just update this line ()

Reply:
Tested this curl is working well
<img width="1586" alt="image" src="https://github.com/gitpod-io/gitpod/assets/20944364/fec458f9-f888-41d4-afb2-df647ee52782">

Reply:
Aaah, that was it, thank you!

Reply:
it's an excellent point about the file not rebuilding itself. I think I'm going to take the approach of pinning it to a specific commit and updating regularly (maybe we can even make a job for it).

Reply:
If you want to make it a job (do you mean GHA?), we could re-use the etag in response header of `https://raw.githubusercontent.com/open-vsx/publish-extensions/master/extension-control/extensions.json`, and let GHA to check if there's anything different then create PR

Reply:
Or better idea maybe, make this etag an arg during leeway build and pass it to this dockerfile. Then leeway can control to update it anyway